python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -952,7 +1008,10 @@ def calculateSpecChiSquare(obsSpectrum, model, radialVelocity, badMask):
     bad = (0 != (obsSpectrum.mask & obsSpectrum.flags.get(*(m for m in badMask if m in obsSpectrum.flags))))
 
     flux = np.copy(obsSpectrum.flux)
-    flux[bad] = 0.0
+    # If not(any(isnan(flux))), we don't have to do this
+    # (because invVar will be set to 0 for bad pixels.)
+    # We set flux[bad] to a finite value to make sure we won't encounter nan.
+    flux[bad] = 1.0
```

Comment:
Are the 'bad' flux values ever used in processing? If so, you shouldn't encounter ``nan`` if only 'good' flux pixels are used.

Reply:
Instead of excluding the bad pixels `flux[bad]`, I set the weights of these pixels to zero. These pixels are actually used (with zero weights). So, these pixels may be any value, but they must be finite. I added the three lines of comment because it took a while to recall what my intention had been here. I changed the r.h.s from 0 to 1 because the unmarked value for the whitened flux is not 0 but 1. When I plot `flux` here, `flux` looks very strange if `flux[bad]` is set to zero.